modules/ballerina-core/src/main/java/org/ballerinalang/util/debugger/VMDebugManager.java

```diff
@@ -325,9 +327,70 @@ public void notifyExit() {
         clientHandler.sendCustomMsg(message);
     }
+
+    /**
+     * Generate debug hit message.
+     *
+     * @param frame             Current stack frame.
+     * @param currentExecLine   Current execution line.
+     * @param threadId          Current thread id.
+     * @return  message         To be sent.
+     */
+    private MessageDTO generateDebugHitMessage(StackFrame frame, LineNumberInfo currentExecLine, String threadId) {
+        MessageDTO message = new MessageDTO(DebugConstants.CODE_HIT, DebugConstants.MSG_HIT);
+        message.setThreadId(threadId);
+
+        BreakPointDTO breakPointDTO = new BreakPointDTO(currentExecLine.getPackageInfo().getPkgPath(),
+                currentExecLine.getFileName(), currentExecLine.getLineNumber());
+        message.setLocation(breakPointDTO);
+
+        int callingIp = currentExecLine.getIp();
+        while (frame != null) {
+            String pck = frame.getPackageInfo().getPkgPath();
+            if (frame.getCallableUnitInfo() == null) {
+                frame = frame.prevStackFrame;
+                continue;
+            }
+            //TODO is this ok to show function name when it's a worker which is executing?
+            String functionName = frame.getCallableUnitInfo().getName();
+            LineNumberInfo callingLine = getLineNumber(frame.getPackageInfo().getPkgPath(), callingIp);
+            FrameDTO frameDTO = new FrameDTO(pck, functionName, callingLine.getFileName(),
+                    callingLine.getLineNumber());
+            message.addFrame(frameDTO);
+            LocalVariableAttributeInfo localVarAttrInfo = (LocalVariableAttributeInfo) frame.getWorkerInfo()
+                    .getAttributeInfo(AttributeInfo.Kind.LOCAL_VARIABLES_ATTRIBUTE);
+            if (localVarAttrInfo == null) {
+                frame = frame.prevStackFrame;
+                continue;
+            }
+            final StackFrame fcp = frame;
+            localVarAttrInfo.getLocalVariables().forEach(localVarInfo -> {
+                VariableDTO variableDTO = new VariableDTO(localVarInfo.getVariableName(), "Local");
+                if (BTypes.typeInt.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BInteger(fcp.getLongLocalVars()[localVarInfo.getVariableIndex()]));
+                } else if (BTypes.typeFloat.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BFloat(fcp.getDoubleLocalVars()[localVarInfo.getVariableIndex()]));
+                } else if (BTypes.typeString.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BString(fcp.getStringLocalVars()[localVarInfo.getVariableIndex()]));
+                } else if (BTypes.typeBoolean.equals(localVarInfo.getVariableType())) {
+                    variableDTO.setBValue(new BBoolean(fcp.getIntLocalVars()[localVarInfo
+                            .getVariableIndex()] == 1 ? true : false));
```

Comment:
Shall we simplify this to ` fcp.getIntLocalVars()[localVarInfo.getVariableIndex()] == 1`